emscripten.py

```diff
@@ -2206,8 +2206,25 @@ def emscript_wasm_backend(infile, outfile, memfile, compiler_engine,
   #   * Run wasm-emscripten-finalize to extract metadata and modify the binary
   #     to use emscripten's wasm<->JS ABI
   #   * Use the metadata to generate the JS glue that goes with the wasm
+  if shared.Settings.MAIN_MODULE:
+    # Run compile_settings here to retrieve all the JS library functions.
+    # Unfortunately, we need to run it later again to capture the exported functions
+    # from the Wasm
+    glue, forwarded_data = compile_settings(compiler_engine, temp_files)
+    forwarded_json = json.loads(forwarded_data)
+    library_fns = forwarded_json['Functions']['libraryFunctions']
+    library_fns_list = []
+    for name in library_fns:
+      library_fns_list.append(name)
+
+    js_symbols = infile + '.js_symbols'
+    with open(js_symbols, 'w') as js_symbols_file:
+      for sym in library_fns_list:
+        js_symbols_file.write(sym + '\n')
+  else:
+    js_symbols = []
```

Comment:
I stole this idea/code and made it part of #10350 in a function called `get_all_js_library_funcs` hopefully that is about to land and we can use that shared function here.